core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -106,42 +109,62 @@ private void waitBeforePreparationSnapshotHookFuture(
     }
   }
 
-  public void commit(Snapshot snapshot) throws CommitException, UnknownTransactionStatusException {
+  public void commit(Snapshot snapshot, boolean readOnly)
+      throws CommitException, UnknownTransactionStatusException {
+    boolean hasSomeWritesOrDeletesInSnapshot = !readOnly && snapshot.hasWritesOrDeletes();
```

Comment:
```suggestion
    boolean hasWritesOrDeletesInSnapshot = !readOnly && snapshot.hasWritesOrDeletes();
```
If `Some` is removed, should it be consistent?

Reply:
Good catch! Fixed in 2c13b1cc50509e8850b506c50fc98883ebafe6ab. Thanks!